components/location/popup.jsx

```diff
@@ -0,0 +1,319 @@
+            </h3>
+          );
+        case "address":
+          return (
+            <address key={i}>
+              <a
+                href={`https://maps.google.com?saddr=Current+Location&daddr=${encodeURI(
+                  location.address
+                )}`}
+                target="_blank"
+                // onClick="captureOutboundLink('${googleMapDirections}', 'directions')"
+                rel="noreferrer"
+              >
+                {location.address}
+              </a>
+            </address>
+          );
+        case "openingForDistributingDonations":
+          return (
+            <OpenHours
+              key={i}
+              closingHours={location.closingForDistributingDonations}
+              id="aid_distribution_hours"
+              openingHours={location.openingForDistributingDonations}
+            />
+          );
+        case "openingForReceivingDonations":
+          return (
+            <OpenHours
+              key={i}
+              closingHours={location.closingForReceivingDonations}
+              id="aid_receiving_hours"
+              openingHours={location.openingForReceivingDonations}
+            />
+          );
+        case "seekingMoney":
+          return <NeedsMoney key={i} location={location} />;
+        case "noIdNeeded":
+          return <NoIdNeeded key={i} location={location} />;
+        case "warmingSite":
+          return <WarmingSite key={i} location={location} />;
+        case "publicTransitOptions":
+          return <PublicTransit key={i} location={location} />;
+        case "accepting":
+          return (
+            <SectionUrl key={i} id="accepting" value={location.accepting} />
+          );
+        case "notAccepting":
+          return (
+            <SectionUrl
+              key={i}
+              id="not_accepting"
+              value={location.notAccepting}
+            />
+          );
+        case "seekingVolunteers":
+          return (
+            <SectionUrl
+              key={i}
+              id="seeking_volunteers_badge"
+              value={location.seekingVolunteers}
+            />
+          );
+        case "mostRecentlyUpdatedAt":
+          return (
+            <div
+              key={i}
+              className="updated-at"
+              title={location.mostRecentlyUpdatedAt}
+            >
+              <span data-translation-id="last_updated">Last updated</span>{" "}
+              <span data-translate-font>
+                {/* TODO: format this! */}
+                {/* ${moment(location.mostRecentlyUpdatedAt, "YYYY-MM-DDTHH:mm:ss.SSSZ").fromNow()} */}
+                {"A million years ago FIX THIS"}
+              </span>
+            </div>
+          );
+        case "urgentNeed":
+          return (
+            <SectionUrl key={i} id="urgent_need" value={location.urgentNeed} />
+          );
+        case "notes":
+          return <SectionUrl key={i} id="notes" value={location.notes} />;
+        default:
```

Comment:
Is the default null or is the default a PopupSection with whatever's passed in? (may not be relevant if all properties are covered above though).

Reply:
I just defaulted this to null but I'm not sure if that's correct honestly! I was having a bit of trouble digesting the existing `propertyTransforms` logic here. I do have a [follow-up](https://github.com/kanadgupta/twin-cities-aid-distribution-locations/blame/8cb738911293f86aa7de13834a6da55e0551b033/issues.md#L29) to make sure that the popup text is identical to the existing one!

Reply:
Haha very understandable, it took me a while to understand too.
Cool, I just double checked the payload coming back from the api and this here has everything except the currentlyOpen which I think you have a note about